src/axom/lumberjack/MPIUtility.cpp

```diff
@@ -49,16 +49,49 @@ const char* mpiBlockingReceiveMessages(MPI_Comm comm)
   return charArray;
 }
 
+const char* mpiNonBlockingReceiveMessages(MPI_Comm comm, const int tag)
+{
+  const int mpiTag = (tag == false) ? LJ_TAG : tag;
+  char* charArray = nullptr;
+  int messageSize = -1;
+  MPI_Status mpiStatus;
+
+  // Get size and source of MPI message
+  int mpiFlag = true;
+  MPI_Iprobe(MPI_ANY_SOURCE, tag, comm, &mpiFlag, &mpiStatus);
```

Comment:
`MPI_Iprobe` is nonblocking here, so is there a chance the `mpiFlag` is not set to `true` when it is expected to be? Would it be better to have this be a blocking `MPI_Probe`? Basing this comment off this stackoverflow post: https://stackoverflow.com/questions/43823458/mpi-iprobe-vs-mpi-probe

Additionally, if using `MPI_Iprobe`, should `mpiFlag` default be set to `false`, so it can be set to `true` only by a successful function call?

Reply:
I think the mpiFlag will be set in either context to either true or false, but to your point, it is safer to initialize this as false.

Reply:
The stackoverflow example illustrates an interesting but slightly different approach than what I'm intending to do.  They are calling MPI_Iprobe in a while loop that does not exit until it returns a flag that is non-zero.  In my case, I am checking to see if any messages need to be received only once, and if there are no messages, the function exits by returning nullptr.  This intent in the stackoverflow example is to continuously monitor the status, whereas I'm only intending to periodically monitor the status whenever the code path enters into this function.  Both could be relevant to the problem I'm trying to solve with this communicator, where the root rank needs to receive information from other ranks that they are aborting. I had a preference toward the latter option (periodically monitoring the status whenever the root rank reaches a point where it enters this code path) because it seemed to me like the more efficient option, even if it comes at a cost of sometimes not receiving the status before the program aborts.  But I'm not really sure which option is best for this scenario.  I'd be curious to hear your thoughts.

Reply:
>  I had a preference toward the latter option (periodically monitoring the status whenever the root rank reaches a point where it enters this code path) because it seemed to me like the more efficient option, even if it comes at a cost of sometimes not receiving the status before the program aborts.

I agree, I would expect the latter option to have less overhead, doing a single poll with `MPI_Iprobe` instead of spinning on `MPI_Iprobe` until status is updated in the former case. Nevertheless, I might not be considering something, so am also curious if others have ideas.